build.gradle.kts

```diff
@@ -341,12 +346,73 @@ tasks {
     return buildCodyDir
   }
 
+  fun copyProtocol(): Unit {
+    // This is just a temporary solution to make it easier to use certain generated protocol
+    // messages.
+    // Ultimately this should entirely be replaced by use of the generated protocol.
+    if (!isForceProtocolCopy) {
+      return
+    }
+    val codyDir = downloadCody()
+    val sourceDir =
+        codyDir.resolve(
+            Paths.get(
+                    "agent",
+                    "bindings",
+                    "kotlin",
+                    "lib",
+                    "src",
+                    "main",
+                    "kotlin",
+                    "com",
+                    "sourcegraph",
+                    "cody",
+                    "protocol_generated")
+                .toString())
+    val targetDir =
+        layout.projectDirectory.asFile.resolve(
+            "src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated")
+
+    targetDir.deleteDirectoryContents()
+    sourceDir.copyRecursively(targetDir, overwrite = true)
+    // in each file replace the package name
+    for (file in targetDir.walkTopDown()) {
+      if (file.isFile && file.extension == "kt") {
+        val content = file.readText()
+        val newContent =
+            """
+        |/*
+        | * Generated file - DO NOT EDIT MANUALLY
+        | * They are copied from the cody agent project using the copyProtocol gradle task.
+        | * This is only a temporary solution before we fully migrate to generated protocol messages.
+        | */
+        |
+    """
+                .trimMargin() +
+                // TODO: these are several "mods" until we can fix the code generation
+                content
+                    // TODO: CODY-2865 replacing ints with longs
+                    .replace(": Int", ": Long")
+                    // replacing classpath
+                    .replace(
+                        "com.sourcegraph.cody.protocol_generated",
```

Comment:
Let's remove these workarounds, it only takes a couple minutes to incorporate these changes in the codegen https://github.com/sourcegraph/cody/pull/4915

Reply:
Yes, I just wanted to have a PR ready to ship the feature as it's time critical. But there's some time left so let's do so.

I might just want to add so we always generate an empty companion object too as it makes extending easier and I can get rid of the factory classes

Reply:
The bindings have been unadopted for ~5 months, is it really time critical?

Reply:
haha; No not in that sense. But my use of the auto bindings in a time critical feature just because I didn't want to dig a deeper hole meant I tried to be cautious about poking at adjacent stuff I didn't fully understand the impact of.

But you have showed me the way.  And properly fixing this in codegen is a lot easier than I thought 😊